frontend/online-shop/src/components/productsList/TableCards.tsx

```diff
@@ -0,0 +1,66 @@
+import { DescriptionCard } from "./DescriptionCard";
+import { useStyle } from "../../styles/productsList/TableCards";
+import { GridList, GridListTile } from "@material-ui/core";
+import { Grid } from "@material-ui/core";
+import { useState } from "react";
+import { useEffect } from "react";
+import { getAll } from "../../services/ProductService";
+
+export const TableCards = (props: any) => {
+  const style = useStyle();
+
+  const [width, setWidth] = useState(window.innerWidth);
+  const [columns, setColumns] = useState(4);
+  const [data, setData] = useState<any[]>([]);
+
+  useEffect(() => {
+    setWidth(window.innerWidth);
+    try {
+      getAll().then((result) => {
+        setData(result.data);
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }, []);
+
+  useEffect(() => {
+    function handleResize() {
+      setWidth(window.innerWidth);
+    }
+    window.addEventListener("resize", handleResize);
+
+    handleResize();
+    setColumns(width < 700 ? 1 : width >= 700 && width < 1100 ? 2 : 3);
+
+    return () => window.removeEventListener("resize", handleResize);
+  }, [width]);
+
+  const tableItems = data.map((product) => (
+    <GridListTile key={product.id.toString()} className={style.tile}>
+      <img src={product.imageUrl} alt={product.name} />
+      <DescriptionCard
+        id={product.id}
```

Comment:
Instead of passing the attributes one by one you could send the whole product at once. `product={product}`